### ISSUE TITLE
AxisError when training the model

### ISSUE THREAD

**Issue by chioveri** [Microsoft/AutonomousDrivingCookbook#19]
Hi all,

I have successfully run and prepared the data associated with the E2EDeepLearning tutorial.  When I try to train the model, however, I get a AxisError (see below). Keras is insalled and all the python modules are up to date (I have run  -installPackages.py). I use the TensorFlow backend. What can be the cause of this error?
Thanks in advance,

-----------------
def draw_image_with_label(img, label, prediction=None):
    theta = label * 0.69 #Steering range for the car is +- 40 degrees -> 0.69 radians
    line_length = 50
    line_thickness = 3
    label_line_color = (255, 0, 0)
    prediction_line_color = (0, 0, 255)
    pil_image = image.array_to_img(img, K.image_data_format(), scale=True)
    print('Actual Steering Angle = {0}'.format(label))
    draw_image = pil_image.copy()
    image_draw = ImageDraw.Draw(draw_image)
    first_point = (int(img.shape[1]/2),img.shape[0])
    second_point = (int((img.shape[1]/2) + (line_length * math.sin(theta))), int(img.shape[0] - (line_length * math.cos(theta))))
    image_draw.line([first_point, second_point], fill=label_line_color, width=line_thickness)
    
    print('Done')
    if (prediction is not None):
        print('Predicted Steering Angle = {0}'.format(prediction))
        print('L1 Error: {0}'.format(abs(prediction-label)))
        theta = prediction * 0.69
        second_point = (int((img.shape[1]/2) + (line_length * math.sin(theta))), int(img.shape[0] - (line_length * math.cos(theta))))
        image_draw.line([first_point, second_point], fill=prediction_line_color, width=line_thickness)
    
    del image_draw
    plt.imshow(draw_image)
    plt.show()

[sample_batch_train_data, sample_batch_test_data] = next(train_generator)
for i in range(0, 3, 1):
    print('Done')
    draw_image_with_label(sample_batch_train_data[0][i], sample_batch_test_data[i])

---------------------------------------------------------------------------
AxisError                                 Traceback (most recent call last)
<ipython-input-9-14df299e308b> in <module>()
     25     plt.show()
     26 
---> 27 [sample_batch_train_data, sample_batch_test_data] = next(train_generator)
     28 for i in range(0, 3, 1):
     29     print('Done')

C:\Users\CHIOVERI\AppData\Local\Continuum\Anaconda3\lib\site-packages\keras\preprocessing\image.py in __next__(self, *args, **kwargs)
    858 
    859     def __next__(self, *args, **kwargs):
--> 860         return self.next(*args, **kwargs)
    861 
    862     def _get_batches_of_transformed_samples(self, index_array):

C:\Users\CHIOVERI\Documents\AutonomousDrivingCookbook-master\AirSimE2EDeepLearning\Generator.py in next(self)
    240         # so it can be done in parallel
    241 
--> 242         return self.__get_indexes(index_array)
    243 
    244     def __get_indexes(self, index_array):

C:\Users\CHIOVERI\Documents\AutonomousDrivingCookbook-master\AirSimE2EDeepLearning\Generator.py in __get_indexes(self, index_array)
    262                 x_images = x_images[self.roi[0]:self.roi[1], self.roi[2]:self.roi[3], :]
    263 
--> 264             transformed = self.image_data_generator.random_transform_with_states(x_images.astype(K.floatx()))
    265             x_images = transformed[0]
    266             is_horiz_flipped.append(transformed[1])

C:\Users\CHIOVERI\Documents\AutonomousDrivingCookbook-master\AirSimE2EDeepLearning\Generator.py in random_transform_with_states(self, x, seed)
    138             x = image.random_channel_shift(x,
    139                                      self.channel_shift_range,
--> 140                                      img_channel_axis)
    141         if self.horizontal_flip:
    142             if np.random.random() < 0.5:

C:\Users\CHIOVERI\AppData\Local\Continuum\Anaconda3\lib\site-packages\keras\preprocessing\image.py in random_channel_shift(x, intensity, channel_axis)
    177 
    178 def random_channel_shift(x, intensity, channel_axis=0):
--> 179     x = np.rollaxis(x, channel_axis, 0)
    180     min_x, max_x = np.min(x), np.max(x)
    181     channel_images = [np.clip(x_channel + np.random.uniform(-intensity, intensity), min_x, max_x)

C:\Users\CHIOVERI\AppData\Local\Continuum\Anaconda3\lib\site-packages\numpy\core\numeric.py in rollaxis(a, axis, start)
   1440     """
   1441     n = a.ndim
-> 1442     axis = normalize_axis_index(axis, n)
   1443     if start < 0:
   1444         start += n

AxisError: axis 3 is out of bounds for array of dimension 3

**Comment by mitchellspryn** [comment]
This looks like a duplicate of [this issue](https://github.com/Microsoft/AutonomousDrivingCookbook/issues/18). What version of Keras is installed? If it's not 2.1.2, try that.
